pkg/flow/adapter/xslttransform/adapter_test.go

```diff
@@ -256,6 +257,67 @@ func TestXSLTTransformEvents(t *testing.T) {
 	}
 }
 
+func TestXSLTTransformKSINK(t *testing.T) {
+	testCases := map[string]struct {
+		allowXSLTOverride bool
+		xslt              string
+
+		inEvent             cloudevents.Event
+		expectedEventString string
+		expectPanic         string
+		expectCategory      string
+	}{
+		"transform ok": {
+			xslt:                tXSLT,
+			inEvent:             newCloudEvent(tXML, cloudevents.ApplicationXML),
+			expectedEventString: "Context Attributes,\n  specversion: 1.0\n  type: ce.test.type\n  source: ce.test.source\n  id: ce-abcd-0123\n  datacontenttype: application/xml\nData (binary),\n  <?xml version=\"1.0\"?>\n<output>\n  <item>A1</item>\n  <item>B2</item>\n  <item>C3</item>\n</output>\n\n",
```

Comment:
That's very fragile. We should check attributes individually, and compare data to some fixed JSON payload, not stringify the entire event like that.
It might break on any update of the CloudEvent SDK. Even a single, unrelated character would break it.

Reply:
Please review my update. I think i got what you are asking for.

Reply:
I didn't even know `assert` was able to perform this kind of nested `struct` comparisons. Looking good 👍